pkg/armrpc/api/v1/types_test.go

```diff
@@ -89,8 +89,9 @@ func TestBaseResource_UpdateMetadata(t *testing.T) {
 
 	armCtx := &ARMRequestContext{Location: "global"}
 	var err error
-	armCtx.ResourceID, err = resources.ParseResource(newResourceID)
+	resourceID, err := resources.ParseResource(newResourceID)
```

Comment:
If this is a good change overall, it might make sense to make `resources.ParseResource` return a pointer. Then you'd be encouraged to use it as a pointer since you'd have to do work to do otherwise.